auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthSecurityConfiguration.java

```diff
@@ -60,31 +63,34 @@ public class VaadinAuthSecurityConfiguration extends VaadinWebSecurity {
 
     private final DefaultVaadinAuthContext vaadinAuthContext;
 
+    private final VaadinUserService vaadinUserService;
+
+    private final SessionRegistry sessionRegistry;
+
+    private final BackChannelLogoutFilter backChannelLogoutFilter;
+
     /**
      * Creates an instance of this configuration bean.
      *
      * @param properties
      *            the configuration properties
+     * @param sessionRegistry
+     *            the session registry
      * @param clientRegistrationRepository
      *            the client-registration repository
      */
     public VaadinAuthSecurityConfiguration(VaadinAuthProperties properties,
+            SessionRegistry sessionRegistry,
             ClientRegistrationRepository clientRegistrationRepository) {
         this.properties = properties;
+        this.sessionRegistry = sessionRegistry;
         this.loginSuccessHandler = new VaadinSavedRequestAwareAuthenticationSuccessHandler();
         this.logoutSuccessHandler = new OidcClientInitiatedLogoutSuccessHandler(
                 clientRegistrationRepository);
         this.vaadinAuthContext = new DefaultVaadinAuthContext();
-    }
-
-    /**
-     * Gets the authentication context bean.
-     *
-     * @return the authentication context bean
-     */
-    @Bean
-    public VaadinAuthContext getAuthContext() {
-        return vaadinAuthContext;
+        this.vaadinUserService = new VaadinUserService();
+        this.backChannelLogoutFilter = new BackChannelLogoutFilter(
```

Comment:
Isn't the component annotation unnecessary if you create the bean yourself? Also, should this only be done if the property is enabled? (Minor or even macro optimization)

Reply:
It's not necessary indeed, that's a leftover from previous prototyping. Good catch, thanks!